app/src/main/java/io/gnosis/kouban/push/PushServiceRepository.kt

```diff
@@ -81,89 +93,140 @@ class PushServiceRepository(
-            val address: String
+            val address: String,
+            val txHash: String
         ) : PushMessage(TYPE) {
             companion object {
                 const val TYPE = "NEW_CONFIRMATION"
                 fun fromMap(params: Map<String, String>) =
                     NewConfirmation(
-                        params.getOrThrow("address")
+                        params.getOrThrow("address"),
+                        params.getOrThrow("txHash")
                     )
             }
         }
 
         data class PendingMultisigTransaction(
-            val address: String
+            val address: String,
+            val txHash: String
         ) : PushMessage(TYPE) {
             companion object {
                 const val TYPE = "PENDING_MULTISIG_TRANSACTION"
                 fun fromMap(params: Map<String, String>) =
                     PendingMultisigTransaction(
-                        params.getOrThrow("address")
+                        params.getOrThrow("address"),
+                        params.getOrThrow("txHash")
                     )
             }
         }
 
         data class ExecutedMultisigTransaction(
-            val address: String
+            val address: String,
+            val txHash: String
         ) : PushMessage(TYPE) {
             companion object {
                 const val TYPE = "EXECUTED_MULTISIG_TRANSACTION"
                 fun fromMap(params: Map<String, String>) =
                     ExecutedMultisigTransaction(
-                        params.getOrThrow("address")
+                        params.getOrThrow("address"),
+                        params.getOrThrow("txHash")
                     )
             }
         }
 
         data class IncomingEther(
-            val address: String
+            val address: String,
+            val txHash: String
         ) : PushMessage(TYPE) {
             companion object {
                 const val TYPE = "INCOMING_ETHER"
                 fun fromMap(params: Map<String, String>) =
                     IncomingEther(
-                        params.getOrThrow("address")
+                        params.getOrThrow("address"),
+                        params.getOrThrow("txHash")
                     )
             }
         }
 
         data class IncomingToken(
-            val address: String
+            val address: String,
+            val txHash: String
         ) : PushMessage(TYPE) {
             companion object {
                 const val TYPE = "INCOMING_TOKEN"
                 fun fromMap(params: Map<String, String>) =
                     IncomingToken(
-                        params.getOrThrow("address")
+                        params.getOrThrow("address"),
+                        params.getOrThrow("txHash")
                     )
             }
         }
 
-
         companion object {
             fun fromMap(params: Map<String, String>) =
                 when (params["type"]) {
```

Comment:
Just a note, I checked out this branch and in `else` of this `when` block we throw, I was thinking this could lead crashing the app if the backend starts sending different types of notifications, in which case I think we should just ignore and not so strictly check. What do you think?

Reply:
In the current implementation not just the else throws but also missing parameters. I think it is fine if you want to ignore the error it should be done in the place where this js called imo

Reply:
https://github.com/gnosis/safe-android-kouban/blob/master/app/src/main/java/io/gnosis/kouban/push/KoubanFirebaseService.kt#L14